src/fides/api/service/connectors/bigquery_connector.py

```diff
@@ -142,22 +137,40 @@ def mask_data(
         privacy_request: PrivacyRequest,
         request_task: RequestTask,
         rows: List[Row],
+        input_data: Optional[Dict[str, List[Any]]] = None,
     ) -> int:
         """Execute a masking request. Returns the number of records updated or deleted"""
+
         query_config = self.query_config(node)
         update_or_delete_ct = 0
         client = self.client()
-        for row in rows:
-            update_or_delete_stmts: List[Executable] = (
-                query_config.generate_masking_stmt(
-                    node, row, policy, privacy_request, client
-                )
-            )
-            if update_or_delete_stmts:
+
+        # Check if we're using DELETE masking strategy
+        if query_config.uses_delete_masking_strategy():
```

Comment:
nit: Is it possible to move this up a level of abstraction, so that _all_ connectors will check for the `delete` strategy, and throw an "unsupported masking strategy" error if we attempt to use it?